packages/web-components/tests/e2e-storybook/cypress/integration/carousel/carousel.e2e.js

```diff
@@ -0,0 +1,343 @@
+    cy.carbonThemesScreenshot({
+      capture: 'viewport',
+    });
+  },
+  checkTextRenders: () => {
+    it('should render card text and arrow icon', () => {
+      cy.get(_selectors.heading).then($heading => {
+        expect($heading).not.to.be.empty;
+      });
+
+      cy.get(_selectors.copy).then($copy => {
+        expect($copy).not.to.be.empty;
+      });
+
+      cy.get(_selectorBase).then($carousel => {
+        if ($carousel.find(_selectors.footer).length > 0) {
+          cy.get(_selectors.footer)
+            .find('svg[slot="icon"] path')
+            .each($icon => {
+              expect($icon).to.have.attr('d', 'M11.8 2.8L10.8 3.8 16.2 9.3 1 9.3 1 10.7 16.2 10.7 10.8 16.2 11.8 17.2 19 10z');
+            });
+        }
+
+        if ($carousel.find(_selectors.videoFooter).length > 0) {
+          cy.get(_selectors.videoFooter)
+            .find('svg[slot="icon"] path')
+            .each($icon => {
+              expect($icon).to.have.attr('d', 'M11.8 2.8L10.8 3.8 16.2 9.3 1 9.3 1 10.7 16.2 10.7 10.8 16.2 11.8 17.2 19 10z');
+            });
+        }
+      });
+
+      cy.takeSnapshots();
+    });
+  },
+  checkImageRenders: () => {
+    it('should render with image', () => {
+      cy.get(_selectors.image).should('be.visible');
+
+      cy.takeSnapshots();
+    });
+  },
+  checkVideoRenders: () => {
+    it('should render the video thumbnail and play button', () => {
+      cy.get(_selectors.video)
+        .shadow()
+        .find('img.bx--image__img')
+        .should('be.visible');
+
+      cy.get(_selectors.video)
+        .find('svg[slot="icon"] path')
+        .each($icon => {
+          expect($icon).to.have.attr(
+            'd',
+            'M26.5555476,43.111135 C26.0032708,43.111135 25.5555476,42.6633959 25.5555476,42.1111111 L25.5555476,20.1111111 C25.5541311,19.7531358 25.7441673,19.4217049 26.0538295,19.2421008 C26.3634918,19.0624967 26.745539,19.0621192 27.0555476,19.2411111 L46.0555476,30.2411111 C46.368866,30.4186723 46.5625038,30.7509842 46.5625038,31.1111111 C46.5625038,31.4712381 46.368866,31.8035499 46.0555476,31.9811111 L27.0555476,42.9811111 C26.9031526,43.0674916 26.7307319,43.1123209 26.5555476,43.111135 Z'
+          );
+        });
+
+      cy.takeSnapshots();
+    });
+  },
+  checkVideoDurationText: () => {
+    it('should render the video duration in the footer', () => {
+      cy.get(_selectors.videoFooter)
+        .find('span.bx--card__cta__copy')
+        .then($duration => {
+          expect($duration).not.to.be.empty;
+        });
+    });
+  },
+  checkSameHeight: () => {
+    it('should have headings all the same height', () => {
+      cy.get(_selectors.heading)
+        .wait(1000)
+        .then($heading => {
+          let headingHeightNum = $heading.first().height();
+          let headingHeight = headingHeightNum + 'px';
+
+          expect($heading).to.not.have.css('height', '0px');
+          expect($heading).to.have.css('height', headingHeight);
```

Comment:
I don't know for sure if this is testing what you think. Since `$heading` is a jQuery object with multiple elements in it, I have a suspicion this only tests the first element, in which case this test only proves that the first element has the same height as itself.

Also, you might consider testing for the headers' computed heights rather than checking for a specific CSS style. This has the benefit make the test implementation agnostic.

Reply:
@jkaeser You were very, very right. I tested by removing the `sameHeight` code from the `carousel.ts` file and sure enough, it passed with the code I originally had. So I added an `each()` to loop through them and check each one's height against the first. It now breaks appropriately when `sameHeight` is missing.

_However_, now it is passing on desktop, but not in mobile. 🤔 Even when I add a long `wait()`, for some reason the `sameHeight` is not getting applied. I could use you and @andy-blum 's suggestions on this. Of note, I'm seeing that the `sameHeight` isn't applying correctly in the scroll check, either (you can see it in the Percy screenshots).

Reply:
IIRC the sameHeight code only causes the same height on objects that are horizontally next to each other. If the elements stack vertically, they likely shouldn't get the height adjustments.

Are there any elements here that are still next to each other at the mobile width?

Reply:
Nope, at mobile the cards are not visible next to each other except when sliding. I'll add a check for the viewport size.